pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1320,7 +1348,10 @@ func (c *AWSCloud) AttachDisk(diskName string, instanceName string, readOnly boo
 
 	err = disk.waitForAttachmentStatus("attached")
 	if err != nil {
-		return "", err
+		// Return the device even if there was an error - it could be time out
+		// and AWS is still trying to attach the volume. Caller may want to
+		// detach it manually.
+		return ec2Device, err
```

Comment:
Indeed, there was one place where `AttachDisk` knows device name and does not return it. Fixed and pushed.